front/src/applications/operationalStudies/components/MacroEditor/utils.ts

```diff
@@ -244,29 +248,98 @@ export const getTrainrunFrequencyFromTimetableItem = (
   return trainrunFrequency;
 };
 
-export const getNetzgrafikColors = () =>
-  Object.entries(NETZGRAFIK_COLOR_PALETTE).map(([colorRef, baseColor], index) => ({
-    id: index,
-    colorRef,
-    color: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
-    colorMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
-    colorRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorDarkMode: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorDarkModeFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
-    colorDarkModeMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
-    colorDarkModeRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-  }));
+export const getNetzgrafikColors = async ({
+  categories,
+  dispatch,
+}: {
+  categories: TrainrunCategory[];
+  dispatch: AppDispatch;
+}) => {
+  const mainColors = Object.entries(NETZGRAFIK_COLOR_PALETTE).map(
+    ([colorRef, baseColor], index) => ({
+      id: index,
+      colorRef,
+      color: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
+      colorMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
+      colorRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorDarkMode: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorDarkModeFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
+      colorDarkModeMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
+      colorDarkModeRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+    })
+  );
+
+  const subCategoryPage = await dispatch(
+    osrdEditoastApi.endpoints.getSubCategory.initiate({ page: 1 }, { subscribe: false })
+  ).unwrap();
 
-export const getTrainMainCategoryFromId = (id: number): TrainMainCategory | null => {
-  for (const key of OSRD_TRAINRUN_CATEGORY_MAPPING.keys()) {
-    if (OSRD_TRAINRUN_CATEGORY_MAPPING.get(key)!.id === id) {
-      return key !== 'NO_CATEGORY' ? key : null;
+  const subCategories: SubCategory[] = [...subCategoryPage.results];
+
+  const subCategoriesFromTrainCategoryToNGE = categories.filter(
+    (_, i) => i > TRAIN_MAIN_CATEGORY_TO_NGE.length - 1
+  );
+
+  const subCategoriesFormatted = subCategoriesFromTrainCategoryToNGE.map((cat) => {
+    const subCategory = subCategories.find((subCat) => subCat.code === cat.colorRef);
+    if (!subCategory) {
+      throw new Error(`SubCategory with code ${cat.colorRef} not found.`);
```

Comment:
```suggestion
  const subCategoriesFormatted = subCategoriesFromTrainCategoryToNGE.map((trainrunCat) => {
    const subCategory = subCategories.find((subCat) => subCat.code === trainrunCat.colorRef);
    if (!subCategory) {
      throw new Error(`SubCategory with code ${trainrunCat.colorRef} not found.`);
```

Nit: same as above, because the OSRD and NGE objects are difficult to keep clearly split, having a preciser name would help IMO (feel free to ignore if it adds too much noise)

Reply:
Done